robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -404,6 +404,23 @@ static Application createApplication(AndroidManifest appManifest, Config config)
         }
       }
 
+      application = ReflectionHelpers.callConstructor(applicationClass);
```

Comment:
Oh cool, this should be the Application that was set in the Manifest and compiled into the arsc table right?

We'll be removing support for legacy resources soon so the block above (getting the application from the appManifest will be removed.

Also, what build systems did you test it with? Gradle / Bazel? With Application modules or Library modules?

Reply:
I'm actually curious how the manifest merger behaves wtf <application name="..."> attributes with app + library modules as well as production vs test manifests.

Reply:
Yes this will pull it from that manifest. I tested on Gradle, both library and application modules.